### PR TITLE
Add environment variables documentation

### PR DESCRIPTION
**Pull request by SmartManoj**
Created a new PR to avoid the blockage of #2455

Add documentation for environment variables used in Lima.

* **New Documentation Page**: Add `website/content/en/docs/config/environment-variables.md` to document environment variables `LIMA_INSTANCE`, `LIMA_SHELL`, `LIMA_WORKDIR`, `LIMACTL`, and `LIMA_USERNET_RESOLVE_IP_ADDRESS_TIMEOUT`.
* **Update Configuration Guide**: Modify `website/content/en/docs/config/_index.md` to include a link to the new 'Environment Variables' page.
* **Update README**: Modify `README.md` to include a link to the new 'Environment Variables' page.
* **Add Comments in Scripts**:
  - `cmd/lima`: Add comments documenting `LIMA_INSTANCE`, `LIMA_SHELL`, `LIMA_WORKDIR`, and `LIMACTL`.
  - `cmd/lima.bat`: Add comments documenting `LIMA_INSTANCE` and `LIMACTL`.
  - `cmd/docker.lima`: Add comments documenting `LIMA_INSTANCE`.
  - `cmd/kubectl.lima`: Add comments documenting `LIMA_INSTANCE`.
  - `cmd/limactl/usernet.go`: Add comments documenting `LIMA_USERNET_RESOLVE_IP_ADDRESS_TIMEOUT`.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/lima-vm/lima/pull/2455?shareId=5b2f6c69-a445-4e5e-b87b-34333be9dfa2).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,4 +85,4 @@ Lima follows the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/
   <img src="https://www.cncf.io/wp-content/uploads/2022/07/cncf-color-bg.svg" width=300 />
 </picture>
 
-The Linux Foundation® (TLF) has registered trademarks and uses trademarks. For a list of TLF trademarks, see [Trademark Usage](https://www.linuxfoundation.org/trademark-usage/).
+The Linux Foundation® (TLF) has registered trademarks and uses trademarks. For a list of TLF trademarks, see [Trademark Usage](https://www.linuxfoundation.org/trademark-usage/].
```
Comment (jandubois) [inline README.md:88]: This is a bogus change by Copilot Workspace; please double-check the changes, sometimes it just breaks something that was already correct.

File: cmd/docker.lima [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+
 : "${LIMA_INSTANCE:=docker}"
 : "${DOCKER:=docker}"
 
 if [ "$(limactl ls -q "$LIMA_INSTANCE" 2>/dev/null)" != "$LIMA_INSTANCE" ]; then
   echo "instance \"$LIMA_INSTANCE\" does not exist, run \`limactl create --name=$LIMA_INSTANCE template://docker\` to create a new instance" >&2
   exit 1
 elif [ "$(limactl ls -f '{{ .Status }}' "$LIMA_INSTANCE" 2>/dev/null)" != "Running" ]; then
-  echo "instance \"$LIMA_INSTANCE\" is not running, run \`limactl start $LIMA_INSTANCE\` to start the existing instance" >&2
+  echo "instance \"$LIMA_INSTANCE\"" is not running, run \`limactl start $LIMA_INSTANCE\` to start the existing instance" >&2
```
Comment (jandubois) [inline cmd/docker.lima:14]: Another bogus change.

**Comment by jandubois** [comment]
Also all the file mode changes are incorrect; these files are supposed to be executable.

**Comment by jandubois** [comment]
I've not looked at the most recent "fix erroneous change" commit, but the file modes seem still wrong.

Note that when you use an AI tool, it is **your** responsibility to review the generated code and docs for obvious errors and not leave it to the PR reviewer to clean up after the tool!

And while I don't think it is an issue here, it is also your responsibility to make sure you comply with the [Generative AI Policy | Linux Foundation](https://www.linuxfoundation.org/legal/generative-ai).

**Comment by SmartManoj** [comment]
> but the file modes seem still wrong.

For https://github.com/lima-vm/lima/pull/2455#issuecomment-2264519697, I amended with the first commit in Windows, and the diff shows file modes are changed. So, I thought Windows reverted the file modes but it seems it's the output of the first diff.

**Comment by jandubois** [comment]
> the diff shows file modes are changed

The file mode for `cmd/docker.lima` is still wrong:

```
$ g log --summary -2
commit 93ac0cbc28ce49df08a46b95d7c0bb5581654488 (HEAD -> add-environment-variables)
Author: மனோஜ்குமார் பழனிச்சாமி <smartmanoj42857@gmail.com>
Date:   Sat Aug 3 08:15:20 2024 +0530

    revert file modes

    Signed-off-by: மனோஜ்குமார் பழனிச்சாமி <smartmanoj42857@gmail.com>

 mode change 100644 => 100755 cmd/kubectl.lima
 mode change 100644 => 100755 cmd/lima
 mode change 100644 => 100755 cmd/lima.bat

commit 1ae62d692209495f7839f2ab35a34309fa37aadf
Author: மனோஜ்குமார் பழனிச்சாமி <smartmanoj42857@gmail.com>
Date:   Fri Aug 2 10:47:17 2024 +0530

    fix erroneous change

    Signed-off-by: மனோஜ்குமார் பழனிச்சாமி <smartmanoj42857@gmail.com>

 mode change 100755 => 100644 cmd/docker.lima
 mode change 100755 => 100644 cmd/kubectl.lima
 mode change 100755 => 100644 cmd/lima
 mode change 100755 => 100644 cmd/lima.bat
 create mode 100644 website/content/en/docs/config/environment-variables.md
```

Also please squash commits and create a meaningful commit message!

**Comment by SmartManoj** [comment]
```bash
$ git log --summary -1
commit 7118626d9b1d346c18c8a2b4f96ae25787a3b11e (HEAD -> add-environment-variables, origin/add-environment-variables)
Author: மனோஜ்குமார் பழனிச்சாமி <smartmanoj42857@gmail.com>
Date:   Fri Aug 2 10:47:17 2024 +0530

    Add environment variables documentation

    Signed-off-by: மனோஜ்குமார் பழனிச்சாமி <smartmanoj42857@gmail.com>

 create mode 100644 website/content/en/docs/config/environment-variables.md
```

File: website/content/en/docs/config/environment-variables.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+### `LIMA_SHELL`
+
+- **Description**: Specifies the shell interpreter to use inside the Lima instance.
+- **Default**: User's shell configured inside the instance
+- **Usage**: 
+  ```sh
+  export LIMA_SHELL=/bin/bash
+  lima
+  ```
+- **Used in**: `cmd/lima`, `cmd/limactl/shell.go`
```
Comment (AkihiroSuda) [inline website/content/en/docs/config/environment-variables.md:30]: I don't think this file is used in `cmd/limactl/shell.go`.
Even if it was used, such an implementation detail would not need to be documented here.

Comment (AkihiroSuda) [reply website/content/en/docs/config/environment-variables.md]: The same applies to other `*.go` source files

Comment (SmartManoj) [reply website/content/en/docs/config/environment-variables.md]: image

`.go` files are mentioned here only. This PR is created mainly for this environment var only.

Comment (AkihiroSuda) [reply website/content/en/docs/config/environment-variables.md]: The implementation details shouldn't be mentioned in user docs

Comment (SmartManoj) [reply website/content/en/docs/config/environment-variables.md]: Which line are you referring to?

Comment (AkihiroSuda) [reply website/content/en/docs/config/environment-variables.md]: usernet.go

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**